quickwit/quickwit-serve/src/lib.rs

```diff
@@ -321,14 +322,19 @@ async fn start_control_plane_if_needed(
             replication_factor,
         )
         .await?;
+        // These layers apply to all the RPCs of the control plane.
+        let shared_layers = ServiceBuilder::new()
+            .layer(CP_GRPC_SERVER_METRICS_LAYER.clone())
+            .layer(LoadShedLayer::new(100))
+            .into_inner();
```

Comment:
should/can we put those after in order to avoid spawning tasks if the load shed kicks in.

Reply:
So [this](https://docs.rs/tower/latest/tower/builder/struct.ServiceBuilder.html#order) is the layering order for `ServiceBuilder`.

The layering order for code-generated `*TowerLayerStack` is the same... Since I couldn't remember it, I wrote a unit test in `codegen-example` (see `test_hello_codegen_tower_stack_layer_ordering`).

So in the end we want:
```rust
.layer(MetricsLayer)
.layer(LoadShedLayer)
.layer(LoadShedLayer)
.layer(OneTaskPerCallLayer)
```
which is what I did here and may have gotten wrong in the past.